src/main/xerus/monstercat/MonsterUtilities.kt

```diff
@@ -310,4 +314,65 @@ class MonsterUtilities(checkForUpdate: Boolean): VBox(), JFXMessageDisplay {
 		}
 	}
 	
+	/** Shows a new window with an ImageView of the requested [coverUrl]
+	 * @param coverUrl URL of the cover to download and show
+	 * @param title Title of the window, only useful when decorated
+	 * @param size Height and width in pixel of the window and image
+	 * @param isDecorated True if the window has borders and title bar with close controls
+	 * @param isDraggable True if the window can be dragged by the mouse
+	 * @param closeOnFocusLost Should we close the window if we're out of focus ?
+	 * @param isResizable Allow resizing the window. The image will follow.
+	 */
+	fun viewCover(coverUrl: String, size: Double? = null, title: String = "Cover Art", isDecorated: Boolean = false, isDraggable: Boolean = true, closeOnFocusLost: Boolean = true, isResizable: Boolean = false){
+		val windowSize: Double = size ?: minOf(Screen.getPrimary().visualBounds.width, Screen.getPrimary().visualBounds.height) / 2
+		
+		val pane = StackPane()
+		val largeImage = ImageView()
+		pane.add(Label("Cover loading..."))
+		pane.add(largeImage)
+		
+		val stage = App.stage.createStage(title, pane).apply {
+			height = windowSize
+			width = windowSize
+			if (isResizable) {
+				widthProperty().addListener { _, _, newValue ->
+					largeImage.fitHeight = newValue as Double
+					largeImage.fitWidth = newValue
+				}
+			}
+			this.isResizable = isResizable
+			
+			widthProperty().addListener { _, _, newValue ->
+				largeImage.fitHeight = newValue as Double
+				largeImage.fitWidth = newValue
+			}
+			
+			initStyle(if (isDecorated) StageStyle.DECORATED else StageStyle.UNDECORATED)
+			
+			if (isDraggable) {
+				var xOffset = 0.0
+				var yOffset = 0.0
+				pane.onMousePressed = EventHandler<MouseEvent> { event ->
+					xOffset = event.sceneX
+					yOffset = event.sceneY
+				}
+				pane.onMouseDragged = EventHandler<MouseEvent> { event ->
+					this.x = event.screenX - xOffset
+					this.y = event.screenY - yOffset
+				}
+			}
+			
+			if (closeOnFocusLost) {
+				focusedProperty().addListener { _, lostFocus, _ ->
```

Comment:
This doesn't look correct. You are calling the old value of focused "lostFocus", but it should be "oldFocus". This might currently work, but is semantically completely incorrect.

Reply:
It's true; if oldFocus is 'true', that means that we are no longer focused; Just replace it with !newFocus

Reply:
Fixed in 1f25196964e7dc2b6a3ef6372af5880f3ec0b6ff

Reply:
Practically it worked like this, but theoretically there could be a case where both oldFocus and newFocus are true, which would make this method falsely trigger. Anyways, it's good now :)